torchbenchmark/util/framework/vision/model_factory.py

```diff
@@ -25,7 +25,7 @@ def __init__(self, model_name, test, device, jit=False, batch_size=None, extra_a
             self.example_outputs = torch.rand_like(self.model(*self.example_inputs))
             self.model.train()
             # setup optimizer and loss_fn
-            self.optimizer = optim.Adam(self.model.parameters())
+            self.optimizer = optim.Adam(self.model.parameters(), capturable=True)
```

Comment:
I'm guessing we probably shouldn't land this because it breaks on non-cuda:

https://github.com/pytorch/pytorch/blob/84c4b079328c2a97d78d95c47b841f8dca6036bb/torch/optim/adam.py#L262

Or maybe we can use `capturable=(device == 'cuda')`. Although maybe this is also bad for non-dynamo use cases (based on Natalia's comment that perf is better when `capturable=False`)

Reply:
yea i'll remove this for now

i guess we'll have to just keep a dev branch while we work out how to properly fix this.

or maybe i'll make this an ENV flag so we can just stick on main branch and workaround till fixed.

Reply:
Will add `ADAM_CAPTURABLE` env flag for this.